roles/image-cache/tasks/main.yml

```diff
@@ -10,8 +10,20 @@
     path: "{{ images_dir }}/{{ image_name }}"
   register: image_path
 
+- name: Check if local path exists and copy to the image-cache directory
+  become: true
+  when: "os_image_path is defined"
+  block:
+  - name: Check if the image exists into the local path
+    stat:
+      path: "{{ os_image_path }}/{{ image_name }}"
```

Comment:
It would be good to default os_image_path in the role defaults so we don't have a hard dependency on the extra_vars (I've been trying to move things into the roles so they can potentially be reused, e.g for stack validation)

Reply:
changed